### PR TITLE
[Enhancement] Tips for optimizing replication_num when creating a table

### LINKED ISSUE #11119: Inconsistent log prompt
### Steps to reproduce the behavior (Required)
CREATE TABLE `tablename` (
  `department_id` int(11) NOT NULL COMMENT ""
) ENGINE=OLAP 
PRIMARY KEY(`department_id`)
DISTRIBUTED BY HASH(`department_id`) BUCKETS 1 
PROPERTIES (
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false"
);

### Expected behavior (Required)
ERROR 1064 (HY000): Failed to find enough host in all backends. need: 3, Current alive backend is [10002]
### Real behavior (Required)
fe.log:
2022-09-13 13:58:19,564 WARN (statistics meta manager|23) [StatisticsMetaManager.createTable():169] Failed to create tableFailed to find enough host in all backends. need: 1, Current alive backend is [10002]，
### StarRocks version (Required)
use 2.3.0 on Docker

### PR DESCRIPTION
**Pull request by Astralidea**
Fixes #11119

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 19 (84.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|3|6|50.00%|[261, 262, 263]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 19 (84.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|3|6|50.00%|[261, 262, 263]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|2|2|100.00%|[]|

**Review by ABingHuang — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

File: fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java [hunk 1]
```diff
@@ -425,9 +426,10 @@ private static void checkReplicationNum(short replicationNum) throws AnalysisExc
             }
         } else {
             if (replicationNum > backendIds.size()) {
-                throw new AnalysisException("Replication num should be less than the number of available BE nodes. "
-                        + "Replication num is " + replicationNum + " available BE nodes is " + backendIds.size() +
-                        ", You can change this default by setting the replication_num table properties.");
+                throw new AnalysisException("Table replication num should be less than " +
+                        "of equal to the number of available BE nodes. "
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java:430]: 1. "than of equal"  --> "than or equal"
2. print the replicationNum  (you removed it)
3. "You can change this default by ..."  -->  "You can change it be ..."
    that, 'default' here is ambiguous.

**Pull request merged**